### ISSUE TITLE
Auto-truncation as a function of HWHM

### ISSUE THREAD

**Issue by erwanp** [radis/radis#612]
### 💭 Description

- [ ] Add an option to set truncation as a number of HWHM (e.g. : 100 HWHM) ;

We currently use `truncation=50 cm-1` by default in any situation, including very low pressure and narrow range spectra, where this is a high penalty.  Lineshape is typically negligeable beyond 1-2 cm-1 in these cases.

**Comment by aditya-pandey-dev** [comment]
Hi @erwanp @minouHub 

I'd like to work on this issue.

Here's my understanding of the problem:
- The current fixed `truncation=50 cm-1` is unnecessarily large for low-pressure spectra where HWHM is typically 0.001–0.05 cm-1, causing a significant performance penalty.
- The fix would involve supporting a string-based `truncation` parameter (e.g., `'auto'` or `'100xHWHM'`) that dynamically sets the truncation as a multiple of the actual Voigt HWHM calculated per line.

My plan:
1. Update `SpectrumFactory` in `factory.py` to accept `truncation` as a float (existing behavior) or a string like `'auto'` / `'NxHWHM'`
2. Add `_resolve_truncation()` logic in `broadening.py` that reads `df1['hwhm_voigt']` after HWHM calculation and computes the truncation in cm-1
3. Keep full backward compatibility (`truncation=50` still works as before)
4. Add tests and a benchmark comparing performance at low vs high pressure

Could you confirm if this approach sounds right, or if there are any design preferences I should follow?

Thanks!

**Comment by minouHub** [comment]
Yes. I would add that this would be applied only below 1 atm for performance purpose. Above 1 atm, let's keep the default truncation

**Comment by minouHub** [comment]
The priority of this issue is set to low because anyone can adjust the truncation factor depending on its conditions. If someone wants line specific truncation factors, they can comment this issue
